### PR TITLE
Make sure canvas size is integral

### PR DESCRIPTION
**Pull request by nmichaud**
When using hypergrid in a layout engine (like phosphor or golden-layout), sometimes these engines give the canvas a non-integral width and height, which causes text aliasing issues since canvas support subpixel referencing. This fix just floors the bounds of the canvas element in a consistent way.

### CODE REVIEW

File: src/lib/Canvas.js [hunk 1]
```diff
@@ -213,16 +213,36 @@ Canvas.prototype = {
         this.stopResizing();
     },
 
+    divBoundingClientRect: function() {
+        // Make sure our canvas has integral dimensions
+        var size = this.div.getBoundingClientRect();
+        var top = Math.floor(rect.top),
+            left = Math.floor(rect.left),
+            width = Math.ceil(rect.width),
+            height = Math.ceil(rect.height);
```
Comment (joneit) [inline src/lib/Canvas.js]: did you try `Math.round` first rather than `.floor` & `.ceil` ?

File: src/lib/Canvas.js [hunk 2]
```diff
@@ -213,16 +213,36 @@ Canvas.prototype = {
...
+
+        return {
+            top: top,
+            right: left+width,
+            bottom: top+height,
```
Comment (joneit) [inline src/lib/Canvas.js]: linter! (spaces required around operators)

File: src/lib/Canvas.js [hunk 3]
```diff
@@ -213,16 +213,36 @@ Canvas.prototype = {
...
+            height: height,
+            x: rect.x,
+            y: rect.y
+        };
+    },
+
     checksize: function() {
         //this is expensive lets do it at some modulo
-        var sizeNow = this.div.getBoundingClientRect();
+        var sizeNow = this.divBoundingClientRect();
```
Comment (joneit) [inline src/lib/Canvas.js]: I'd prefer methods names to be verbal, please.
So in this case prefix with `get` or `calc` , _e.g.,_ `getDivRect`

File: src/lib/Canvas.js (hunk 1 above)
Comment (nmichaud) [reply src/lib/Canvas.js]: Round isn't quite right - i need to always round down for the top and left, and always round up for the bottom and right.

**Comment by dcchuck** [comment]
👍

**Comment by joneit** [comment]
@dcchuck thanks for the thumbs-up. Going forward, please approve by:

1. go to …/655/files (or click Files Changed tab)
2. click Review Changes button
3. click Approved radio button 
4. click Submit Review button 

This way it will say “Approved” on the pull request list.
 Thanks!

**Comment by joneit** [comment]
@nmichaud we got our first 2-team-approved PR. merging now...

**Pull request merged**